container_registry/container_analysis/samples.py

```diff
@@ -0,0 +1,343 @@
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    return client.get_occurrence(occurrence_name)
+# [END get_occurrence]
+
+
+# [START discovery_info]
+def get_discovery_info(image_url, project_id):
+    """prints the Discovery occurrence created for a specified image
+    This occurrence contains information about the initial scan on the image
+
+    :param image_url:  the Container Registry URL associated with the image
+                example: "https://gcr.io/project/image@sha256:foo"
+    :param project_id: the GCP project the occurrence will be created under
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    filterStr = "kind=\"DISCOVERY\" AND resourceUrl=\"" + image_url + "\""
+    client = container_analysis_client.ContainerAnalysisClient()
+    project_name = client.project_path(project_id)
+    response = client.list_occurrences(project_name, filter_=filterStr)
+    for occ in response:
+        print occ
+# [END discovery_info]
+
+
+# [START occurrences_for_note]
+def get_occurrences_for_note(note_id, project_id):
+    """Retrieves all the occurrences associated with a specified note
+
+    :param note_id: the note's unique identifier
+    :param project_id: the project's unique identifier
+    :return: number of occurrences found
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    note_name = client.note_path(project_id, note_id)
+
+    response = client.list_note_occurrences(note_name)
+    count = 0
+    for o in response:
+        # do something with the retrieved occurrence
+        # in this sample, we will simply count each one
+        count += 1
+    return count
+# [END occurrences_for_note]
+
+
+# [START occurrences_for_image]
+def get_occurrences_for_image(image_url, project_id):
+    """Retrieves all the occurrences associated with a specified image
+
+    :param note_id: the note's unique identifier
+    :param project_id: the project's unique identifier
+    :return: number of occurrences found
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    filterStr = "resourceUrl=\"" + image_url + "\""
```

Comment:
1. Prefer format over string concatenation
2. snake_case and not camelCase
3. Single-quotes over double-quotes

e.g. 

    filter_str = 'resourceUrl="{}".format(image_url)'